plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/edit/index.tsx

```diff
@@ -69,7 +69,9 @@ const AddToCartOptionsEdit = ( props: BlockEditProps< Attributes > ) => {
 						<button
 							className={ `alt wp-element-button ${ productType }_add_to_cart_button` }
 						>
-							{ __( 'Add to cart', 'woocommerce' ) }
+							{ product && isExternalProduct( product )
+								? product.button_text
+								: __( 'Add to cart', 'woocommerce' ) }
```

Comment:
Here we should also check for `product?.add_to_cart?.single_text`. Otherwise, the button text is not respected for third party product types.

Steps to reproduce:

1. Add this code snippet:

```PHP
add_filter( 'woocommerce_product_single_add_to_cart_text', function() {
	return 'Button text in the product page';
});
```
2. Create a Composite product (or any 3rd-party product type).
3. Create a post or page and add the _Product_ block choosing the Composite product you just created.
4. In this branch, `Add to cart` is displayed, when it should be `Button text in the product page`.

Reply:
Good point. I replace it with this code:

https://github.com/woocommerce/woocommerce/blob/60ab5cac36aa6f4993bc8b992fb93db9cfd7d97d/plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/edit/index.tsx#L72-L73